DNN Platform/Library/Common/Utilities/UrlUtils.cs

```diff
@@ -174,7 +174,8 @@ public static string[] GetQSParamsForNavigateURL()
         /// <returns>true if HTTPS or if HTTP with an SSL offload header value, false otherwise</returns>
         public static bool IsSecureConnectionOrSslOffload(HttpRequest request)
         {
-            if (request.IsSecureConnection)
+            var isRequestSSLOffloaded = IsRequestSSLOffloaded(request);
+            if (request.IsSecureConnection || isRequestSSLOffloaded)
```

Comment:
SSL Offloading header in load balancing  is equal to asp.net request.IsSecureConnection. Mean while this check is enough and below all code can be delete, which redundantly checking for page.IsSecure

Reply:
I do not follow. If this was enough to check `HttpRequest#IsSecureConnection` then these change would not be necessary, right? 😉 

After all, with SSL offloading the request coming to the IIS is no longer encrypted, hence the flag would be `false`.